{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas\n",
    "import requests\n",
    "from Bio import Entrez\n",
    "from Bio.KEGG import REST\n",
    "from Bio import SeqIO"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Accession</th>\n",
       "      <th>Begin</th>\n",
       "      <th>End</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Orientation</th>\n",
       "      <th>Name</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Gene ID</th>\n",
       "      <th>Gene Type</th>\n",
       "      <th>Transcripts accession</th>\n",
       "      <th>Protein accession</th>\n",
       "      <th>Protein length</th>\n",
       "      <th>Locus tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NC_001133.9</td>\n",
       "      <td>1807</td>\n",
       "      <td>2169</td>\n",
       "      <td>I</td>\n",
       "      <td>minus</td>\n",
       "      <td>seripauperin PAU8</td>\n",
       "      <td>PAU8</td>\n",
       "      <td>851229</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NM_001180043.1</td>\n",
       "      <td>NP_009332.1</td>\n",
       "      <td>120.0</td>\n",
       "      <td>YAL068C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NC_001133.9</td>\n",
       "      <td>2480</td>\n",
       "      <td>2707</td>\n",
       "      <td>I</td>\n",
       "      <td>plus</td>\n",
       "      <td>uncharacterized protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1466426</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NM_001184582.1</td>\n",
       "      <td>NP_878038.1</td>\n",
       "      <td>75.0</td>\n",
       "      <td>YAL067W-A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>NC_001133.9</td>\n",
       "      <td>7235</td>\n",
       "      <td>9016</td>\n",
       "      <td>I</td>\n",
       "      <td>minus</td>\n",
       "      <td>putative permease SEO1</td>\n",
       "      <td>SEO1</td>\n",
       "      <td>851230</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NM_001178208.1</td>\n",
       "      <td>NP_009333.1</td>\n",
       "      <td>593.0</td>\n",
       "      <td>YAL067C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>NC_001133.9</td>\n",
       "      <td>11565</td>\n",
       "      <td>11951</td>\n",
       "      <td>I</td>\n",
       "      <td>minus</td>\n",
       "      <td>uncharacterized protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>851232</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NM_001179897.1</td>\n",
       "      <td>NP_009335.1</td>\n",
       "      <td>128.0</td>\n",
       "      <td>YAL065C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>NC_001133.9</td>\n",
       "      <td>12046</td>\n",
       "      <td>12426</td>\n",
       "      <td>I</td>\n",
       "      <td>plus</td>\n",
       "      <td>uncharacterized protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>851233</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NM_001180042.1</td>\n",
       "      <td>NP_009336.1</td>\n",
       "      <td>126.0</td>\n",
       "      <td>YAL064W-B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6443</th>\n",
       "      <td>NC_001224.1</td>\n",
       "      <td>48901</td>\n",
       "      <td>50097</td>\n",
       "      <td>MT</td>\n",
       "      <td>plus</td>\n",
       "      <td>mitochondrial 37S ribosomal protein VAR1</td>\n",
       "      <td>VAR1</td>\n",
       "      <td>854586</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_009320.1</td>\n",
       "      <td>398.0</td>\n",
       "      <td>Q0140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6445</th>\n",
       "      <td>NC_001224.1</td>\n",
       "      <td>61022</td>\n",
       "      <td>61729</td>\n",
       "      <td>MT</td>\n",
       "      <td>plus</td>\n",
       "      <td>intron-encoded endonuclease I-SceI</td>\n",
       "      <td>SCEI</td>\n",
       "      <td>854590</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_009324.1</td>\n",
       "      <td>235.0</td>\n",
       "      <td>Q0160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6462</th>\n",
       "      <td>NC_001224.1</td>\n",
       "      <td>73758</td>\n",
       "      <td>74513</td>\n",
       "      <td>MT</td>\n",
       "      <td>plus</td>\n",
       "      <td>cytochrome c oxidase subunit 2</td>\n",
       "      <td>COX2</td>\n",
       "      <td>854622</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_009326.1</td>\n",
       "      <td>251.0</td>\n",
       "      <td>Q0250</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6463</th>\n",
       "      <td>NC_001224.1</td>\n",
       "      <td>74495</td>\n",
       "      <td>75984</td>\n",
       "      <td>MT</td>\n",
       "      <td>plus</td>\n",
       "      <td>maturase-like protein</td>\n",
       "      <td>NaN</td>\n",
       "      <td>854623</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_009327.1</td>\n",
       "      <td>472.0</td>\n",
       "      <td>Q0255</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6467</th>\n",
       "      <td>NC_001224.1</td>\n",
       "      <td>79213</td>\n",
       "      <td>80022</td>\n",
       "      <td>MT</td>\n",
       "      <td>plus</td>\n",
       "      <td>cytochrome c oxidase subunit 3</td>\n",
       "      <td>COX3</td>\n",
       "      <td>854627</td>\n",
       "      <td>protein-coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NP_009328.1</td>\n",
       "      <td>269.0</td>\n",
       "      <td>Q0275</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6014 rows Ã— 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Accession  Begin    End Chromosome Orientation  \\\n",
       "0     NC_001133.9   1807   2169          I       minus   \n",
       "1     NC_001133.9   2480   2707          I        plus   \n",
       "2     NC_001133.9   7235   9016          I       minus   \n",
       "3     NC_001133.9  11565  11951          I       minus   \n",
       "4     NC_001133.9  12046  12426          I        plus   \n",
       "...           ...    ...    ...        ...         ...   \n",
       "6443  NC_001224.1  48901  50097         MT        plus   \n",
       "6445  NC_001224.1  61022  61729         MT        plus   \n",
       "6462  NC_001224.1  73758  74513         MT        plus   \n",
       "6463  NC_001224.1  74495  75984         MT        plus   \n",
       "6467  NC_001224.1  79213  80022         MT        plus   \n",
       "\n",
       "                                          Name Symbol  Gene ID  \\\n",
       "0                            seripauperin PAU8   PAU8   851229   \n",
       "1                      uncharacterized protein    NaN  1466426   \n",
       "2                       putative permease SEO1   SEO1   851230   \n",
       "3                      uncharacterized protein    NaN   851232   \n",
       "4                      uncharacterized protein    NaN   851233   \n",
       "...                                        ...    ...      ...   \n",
       "6443  mitochondrial 37S ribosomal protein VAR1   VAR1   854586   \n",
       "6445        intron-encoded endonuclease I-SceI   SCEI   854590   \n",
       "6462            cytochrome c oxidase subunit 2   COX2   854622   \n",
       "6463                     maturase-like protein    NaN   854623   \n",
       "6467            cytochrome c oxidase subunit 3   COX3   854627   \n",
       "\n",
       "           Gene Type Transcripts accession Protein accession  Protein length  \\\n",
       "0     protein-coding        NM_001180043.1       NP_009332.1           120.0   \n",
       "1     protein-coding        NM_001184582.1       NP_878038.1            75.0   \n",
       "2     protein-coding        NM_001178208.1       NP_009333.1           593.0   \n",
       "3     protein-coding        NM_001179897.1       NP_009335.1           128.0   \n",
       "4     protein-coding        NM_001180042.1       NP_009336.1           126.0   \n",
       "...              ...                   ...               ...             ...   \n",
       "6443  protein-coding                   NaN       NP_009320.1           398.0   \n",
       "6445  protein-coding                   NaN       NP_009324.1           235.0   \n",
       "6462  protein-coding                   NaN       NP_009326.1           251.0   \n",
       "6463  protein-coding                   NaN       NP_009327.1           472.0   \n",
       "6467  protein-coding                   NaN       NP_009328.1           269.0   \n",
       "\n",
       "      Locus tag  \n",
       "0       YAL068C  \n",
       "1     YAL067W-A  \n",
       "2       YAL067C  \n",
       "3       YAL065C  \n",
       "4     YAL064W-B  \n",
       "...         ...  \n",
       "6443      Q0140  \n",
       "6445      Q0160  \n",
       "6462      Q0250  \n",
       "6463      Q0255  \n",
       "6467      Q0275  \n",
       "\n",
       "[6014 rows x 13 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "Entrez.email = \"ppi_project@protonmail.me\"\n",
    "transcriptome = pandas.read_csv(\"Data/saccharomyces_cerevisiae.tsv\", sep=\"\\t\")\n",
    "proteome = transcriptome.loc[transcriptome[\"Gene Type\"] == \"protein-coding\"]\n",
    "proteome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "for acc in proteome[\"Protein accession\"]:\n",
    "    with open(f\"Data/Sequence/{acc}.txt\", \"w\") as filout:\n",
    "        filout.write(Entrez.efetch(db=\"protein\", id=acc, rettype=\"fasta\", retmode=\"text\").read())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open(\"Data/pathways.txt\", \"r\") as fil:\n",
    "    pathways = list(map(lambda x: x.split(\"\\t\"), fil.readlines()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "YLL039C\n",
      "YIL148W\n",
      "YKR094C\n",
      "YLR167W\n",
      "YKL210W\n",
      "YPR180W\n",
      "YDR390C\n",
      "YPR066W\n",
      "YGL058W\n",
      "YOR339C\n",
      "YDR059C\n",
      "YBR082C\n",
      "YMR022W\n",
      "YEL012W\n",
      "YDL064W\n",
      "YER100W\n",
      "YDR177W\n",
      "YLR306W\n",
      "YDR092W\n",
      "YDR054C\n",
      "YGL141W\n",
      "YKL010C\n",
      "YER125W\n",
      "YDR457W\n",
      "YDL190C\n",
      "YLL036C\n",
      "YDR409W\n",
      "YOL013C\n",
      "YOL133W\n",
      "YDL132W\n",
      "YDR328C\n",
      "YFL009W\n",
      "YIL046W\n",
      "YJR090C\n",
      "YPL046C\n",
      "YGR003W\n",
      "YDR030C\n",
      "YDL008W\n",
      "YLR127C\n",
      "YGL116W\n",
      "YGL003C\n",
      "YNL172W\n",
      "YBL084C\n",
      "YDR118W\n",
      "YOR249C\n",
      "YKL022C\n",
      "YHR166C\n",
      "YLR102C\n",
      "YGL240W\n",
      "YFR036W\n",
      "YDR260C\n"
     ]
    }
   ],
   "source": [
    "for pt in pathways:\n",
    "    with open(f\"Data/Pathways/{pt[1]}.txt\", \"w\") as filout:\n",
    "        texte = REST.kegg_get(pt[0]).read()\n",
    "        if \"GENE\" in texte:\n",
    "            texte = texte[texte.find(\"GENE\")+4:]\n",
    "            if \"COMPOUND\" in texte:\n",
    "                texte = texte[:texte.find(\"COMPOUND\")-1].split(\"\\n\")\n",
    "            else:\n",
    "                texte = texte[:texte.find(\"REFERENCE\")-1].split(\"\\n\")\n",
    "            t = '\\n'.join(list(map(lambda x: x.split()[0], texte)))\n",
    "        filout.write(t)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[['sce01100', 'Metabolic pathways \\n'],\n",
       " ['sce01110', 'Biosynthesis of secondary metabolites \\n'],\n",
       " ['sce01200', 'Carbon metabolism \\n'],\n",
       " ['sce01210', '2-Oxocarboxylic acid metabolism\\n'],\n",
       " ['sce01212', 'Fatty acid metabolism \\n'],\n",
       " ['sce01230', 'Biosynthesis of amino acids \\n'],\n",
       " ['sce01232', 'Nucleotide metabolism \\n'],\n",
       " ['sce01250', 'Biosynthesis of nucleotide sugars \\n'],\n",
       " ['sce01240', 'Biosynthesis of cofactors \\n'],\n",
       " ['sce00010', 'Glycolysis / Gluconeogenesis \\n'],\n",
       " ['sce00020', 'Citrate cycle (TCA cycle) \\n'],\n",
       " ['sce00030', 'Pentose phosphate pathway \\n'],\n",
       " ['sce00040', 'Pentose and glucuronate interconversions \\n'],\n",
       " ['sce00051', 'Fructose and mannose metabolism \\n'],\n",
       " ['sce00052', 'Galactose metabolism \\n'],\n",
       " ['sce00053', 'Ascorbate and aldarate metabolism \\n'],\n",
       " ['sce00500', 'Starch and sucrose metabolism \\n'],\n",
       " ['sce00520', 'Amino sugar and nucleotide sugar metabolism \\n'],\n",
       " ['sce00620', 'Pyruvate metabolism \\n'],\n",
       " ['sce00630', 'Glyoxylate and dicarboxylate metabolism \\n'],\n",
       " ['sce00640', 'Propanoate metabolism \\n'],\n",
       " ['sce00650', 'Butanoate metabolism \\n'],\n",
       " ['sce00660', 'C5-Branched dibasic acid metabolism\\n'],\n",
       " ['sce00562', 'Inositol phosphate metabolism \\n'],\n",
       " ['sce00190', 'Oxidative phosphorylation \\n'],\n",
       " ['sce00680', 'Methane metabolism \\n'],\n",
       " ['sce00910', 'Nitrogen metabolism \\n'],\n",
       " ['sce00920', 'Sulfur metabolism \\n'],\n",
       " ['sce00061', 'Fatty acid biosynthesis \\n'],\n",
       " ['sce00062', 'Fatty acid elongation \\n'],\n",
       " ['sce00071', 'Fatty acid degradation \\n'],\n",
       " ['sce00100', 'Steroid biosynthesis \\n'],\n",
       " ['sce00561', 'Glycerolipid metabolism \\n'],\n",
       " ['sce00564', 'Glycerophospholipid metabolism \\n'],\n",
       " ['sce00565', 'Ether lipid metabolism \\n'],\n",
       " ['sce00600', 'Sphingolipid metabolism \\n'],\n",
       " ['sce00590', 'Arachidonic acid metabolism \\n'],\n",
       " ['sce00592', 'alpha-Linolenic acid metabolism\\n'],\n",
       " ['sce01040', 'Biosynthesis of unsaturated fatty acids \\n'],\n",
       " ['sce00230', 'Purine metabolism \\n'],\n",
       " ['sce00240', 'Pyrimidine metabolism \\n'],\n",
       " ['sce00250', 'Alanine, aspartate and glutamate metabolism \\n'],\n",
       " ['sce00260', 'Glycine, serine and threonine metabolism \\n'],\n",
       " ['sce00270', 'Cysteine and methionine metabolism \\n'],\n",
       " ['sce00280', 'Valine, leucine and isoleucine degradation \\n'],\n",
       " ['sce00290', 'Valine, leucine and isoleucine biosynthesis \\n'],\n",
       " ['sce00300', 'Lysine biosynthesis \\n'],\n",
       " ['sce00310', 'Lysine degradation \\n'],\n",
       " ['sce00220', 'Arginine biosynthesis \\n'],\n",
       " ['sce00330', 'Arginine and proline metabolism \\n'],\n",
       " ['sce00340', 'Histidine metabolism \\n'],\n",
       " ['sce00350', 'Tyrosine metabolism \\n'],\n",
       " ['sce00360', 'Phenylalanine metabolism \\n'],\n",
       " ['sce00380', 'Tryptophan metabolism \\n'],\n",
       " ['sce00400', 'Phenylalanine, tyrosine and tryptophan biosynthesis \\n'],\n",
       " ['sce00410', 'beta-Alanine metabolism\\n'],\n",
       " ['sce00430', 'Taurine and hypotaurine metabolism \\n'],\n",
       " ['sce00440', 'Phosphonate and phosphinate metabolism \\n'],\n",
       " ['sce00450', 'Selenocompound metabolism \\n'],\n",
       " ['sce00460', 'Cyanoamino acid metabolism \\n'],\n",
       " ['sce00470', 'D-Amino acid metabolism\\n'],\n",
       " ['sce00480', 'Glutathione metabolism \\n'],\n",
       " ['sce00510', 'N-Glycan biosynthesis\\n'],\n",
       " ['sce00513', 'Various types of N-Glycan biosynthesis\\n'],\n",
       " ['sce00515', 'Mannose type O-glycan biosynthesis\\n'],\n",
       " ['sce00514', 'Other types of O-glycan biosynthesis\\n'],\n",
       " ['sce00563', 'Glycosylphosphatidylinositol (GPI) biosynthesis\\n'],\n",
       " ['sce00511', 'Other glycan degradation \\n'],\n",
       " ['sce00730', 'Thiamine metabolism \\n'],\n",
       " ['sce00740', 'Riboflavin metabolism \\n'],\n",
       " ['sce00750', 'Vitamin B6 metabolism \\n'],\n",
       " ['sce00760', 'Nicotinate and nicotinamide metabolism \\n'],\n",
       " ['sce00770', 'Pantothenate and CoA biosynthesis \\n'],\n",
       " ['sce00780', 'Biotin metabolism \\n'],\n",
       " ['sce00785', 'Lipoic acid metabolism \\n'],\n",
       " ['sce00790', 'Folate biosynthesis \\n'],\n",
       " ['sce00670', 'One carbon pool by folate \\n'],\n",
       " ['sce00860', 'Porphyrin metabolism \\n'],\n",
       " ['sce00130', 'Ubiquinone and other terpenoid biosynthesis\\n'],\n",
       " ['sce00900', 'Terpenoid backbone biosynthesis \\n'],\n",
       " ['sce00909', 'Sesquiterpenoid and triterpenoid biosynthesis \\n'],\n",
       " ['sce00332', 'Carbapenem biosynthesis \\n'],\n",
       " ['sce00261', 'Monobactam biosynthesis \\n'],\n",
       " ['sce00999', 'Biosynthesis of various plant secondary metabolites \\n'],\n",
       " ['sce00791', 'Atrazine degradation \\n'],\n",
       " ['sce03020', 'RNA polymerase \\n'],\n",
       " ['sce03022', 'Basal transcription factors \\n'],\n",
       " ['sce03040', 'Spliceosome \\n'],\n",
       " ['sce03010', 'Ribosome \\n'],\n",
       " ['sce00970', 'Aminoacyl-tRNA biosynthesis\\n'],\n",
       " ['sce03013', 'Nucleocytoplasmic transport \\n'],\n",
       " ['sce03015', 'mRNA surveillance pathway \\n'],\n",
       " ['sce03008', 'Ribosome biogenesis in eukaryotes \\n'],\n",
       " ['sce03060', 'Protein export \\n'],\n",
       " ['sce04141', 'Protein processing in endoplasmic reticulum \\n'],\n",
       " ['sce04130', 'SNARE interactions in vesicular transport \\n'],\n",
       " ['sce04120', 'Ubiquitin mediated proteolysis \\n'],\n",
       " ['sce04122', 'Sulfur relay system \\n'],\n",
       " ['sce03050', 'Proteasome \\n'],\n",
       " ['sce03018', 'RNA degradation \\n'],\n",
       " ['sce03030', 'DNA replication \\n'],\n",
       " ['sce03410', 'Base excision repair \\n'],\n",
       " ['sce03420', 'Nucleotide excision repair \\n'],\n",
       " ['sce03430', 'Mismatch repair \\n'],\n",
       " ['sce03440', 'Homologous recombination \\n'],\n",
       " ['sce03450', 'Non-homologous end-joining\\n'],\n",
       " ['sce03082', 'ATP-dependent chromatin remodeling\\n'],\n",
       " ['sce03083', 'Polycomb repressive complex \\n'],\n",
       " ['sce03250', 'Viral life cycle \\n'],\n",
       " ['sce02010', 'ABC transporters \\n'],\n",
       " ['sce04011', 'MAPK signaling pathway \\n'],\n",
       " ['sce04392', 'Hippo signaling pathway \\n'],\n",
       " ['sce04070', 'Phosphatidylinositol signaling system \\n'],\n",
       " ['sce04144', 'Endocytosis \\n'],\n",
       " ['sce04145', 'Phagosome \\n'],\n",
       " ['sce04146', 'Peroxisome \\n'],\n",
       " ['sce04138', 'Autophagy - yeast\\n'],\n",
       " ['sce04136', 'Autophagy - other\\n'],\n",
       " ['sce04139', 'Mitophagy \\n'],\n",
       " ['sce04148', 'Efferocytosis \\n'],\n",
       " ['sce04111', 'Cell cycle \\n'],\n",
       " ['sce04113', 'Meiosis \\n'],\n",
       " ['sce04814', 'Motor proteins \\n'],\n",
       " ['sce04820', 'Cytoskeleton in muscle cells \\n'],\n",
       " ['sce04213', 'Longevity regulating pathway \\n']]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
